pkg/apis/certmanager/v1alpha1/types.go

```diff
@@ -29,6 +29,21 @@ const (
 // Annotation names for CertificateRequests
 const (
 	CRPrivateKeyAnnotationKey = "certmanager.k8s.io/private-key-secret-name"
+
+	// If this annotation is specified on a Certificate or Order resource when
+	// using the HTTP01 solver type, the ingress.name field of the HTTP01
+	// solver's configuration will be set to the value given here.
+	// This is especially useful for users of Ingress controllers that maintain
+	// a 1:1 mapping between endpoint IP and Ingress resource.
+	ACMECertificateHTTP01IngressNameOverride = "acme.certmanager.k8s.io/http01-override-ingress-name"
+
+	// If this annotation is specified on a Certificate or Order resource when
+	// using the HTTP01 solver type, the ingress.class field of the HTTP01
+	// solver's configuration will be set to the value given here.
+	// This is especially useful for users deploying many different ingress
+	// classes into a single cluster that want to be able to re-use a single
+	// solver for each ingress class.
+	ACMECertificateHTTP01IngressClassOverride = "acme.certmanager.k8s.io/http01-override-ingress-class"
```

Comment:
@JoshVanL we aren't particularly consistent with our prefixes for group names (i.e. `acme.example.com` vs `example.com/acme-*`), but I feel like it makes sense to namespace these (and probably move towards a model like this in future too if we 'break up the monolith' with the Issuer resource)

Reply:
Yep, I agree